src/main/java/jimmy/mcgymmy/logic/commands/EditCommand.java

```diff
@@ -111,8 +112,11 @@ public CommandResult execute(Model model) throws CommandException {
 
         Food editedFood = new Food(newName, newProtein, newFat, newCarb, tags, newDate);
 
+        if (foodToEdit.equals(editedFood)) {
+            return new CommandResult(String.format(MESSAGE_FOOD_NO_CHANGE, editedFood));
+        }
+
         model.setFood(index, editedFood);
-        model.updateFilteredFoodList(Model.PREDICATE_SHOW_ALL_FOODS);
```

Comment:
Is this for it keeping the same filter after the edit of the food?

Reply:
I move those lines in EditCommand, TagCommand, and UnTagCommand `ModelManager#setFood` instead since they all call this method